### PR TITLE
feat: Transaction history components

### PR DESCRIPTION
**Pull request by ryanio**
This PR creates the TxHistory component from Mist with a list of comprehensive stories:

screen shot 2019-01-03 at 7 37 41 am

screen shot 2019-01-03 at 7 37 55 am

### CODE REVIEW

File: src/components/Tx/TxHistory/TxEntry.jsx [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+
+  div {
+    margin-bottom: 10px;
+  }
+
+  a {
+    text-decoration: none;
+  }
+
+  .moreDetails {
```
Comment (marcgarreau) [inline src/components/Tx/TxHistory/TxEntry.jsx]: I'm still unsure of best practices in styled-components. Couple open-ended questions:
1. Should a styled component contain many children styles (like you've done here), or should each of those child elements be styled as separate JS objects? (I've been doing the latter)
2) How much style configurability should we permit? We're close to "none" on the spectrum and maybe that's okay.

File: src/components/Tx/TxHistory/TxEntry.jsx [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+            <div className="label">{i18n.t('mist.txHistory.to')}</div>
+            <div className="identicon">
+              <Identicon address={tokensTo} size="small" />
+            </div>
+            <span className="value">{tokensTo}</span>
+          </div>
+        )}
+        {!isTokenTransfer && (
+          <div>
+            {tx.to && (
```
Comment (marcgarreau) [inline src/components/Tx/TxHistory/TxEntry.jsx]: I think this component can be broken down into a few sub-components. It's a bit unfriendly to read, because of all the conditional logic forks. Consider each fork as a potential child component refactor. Could be a separate PR if you prefer.

File: src/components/Tx/TxHistory/TxEntry.jsx [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+    super(props)
+
+    this.state = { showDetails: false }
+  }
+
+  toggleDetails() {
+    this.setState(prevState => ({ showDetails: !prevState.showDetails }))
+  }
+
+  renderDetails() {
```
Comment (marcgarreau) [inline src/components/Tx/TxHistory/TxEntry.jsx]: Lot packed in there; potential component.

File: src/components/index.js [hunk 4]
```diff
@@ -48,10 +48,10 @@ export {
   NetworkStatus,
   NodeInfo,
   NodeInfoBox,
+  TxHistory,
   Pulse,
   RpcTester,
   Spinner,
   TokenListForItem,
-  TxHistory,
```
Comment (marcgarreau) [inline src/components/index.js]: Not a strong preference, but these were alphabetized.

File: src/lib/txHistory.import.less [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    margin-bottom: 0;
+  }
+}
+
+.no-txs {
+  font-style: italic;
+  font-weight: bold;
+  margin-top: 5px;
+  color: #333;
+}
```
Comment (marcgarreau) [inline src/lib/txHistory.import.less]: Would prefer not to import any more `.less` files now that we're in styled-components land. Don't want to have to track down styles in multiple places.

File: src/components/Tx/TxHistory/TxEntry.jsx [hunk 6]
```diff
@@ -0,0 +1,349 @@
+import React, { Component } from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import web3 from 'web3'
+import i18n from '../../../i18n'
+import Identicon from '../../Identicon'
+import { networkIdToName, hexToNumberString } from '../../../lib/util'
+
+const BN = web3.utils.BN
```
Comment (marcgarreau) [inline src/components/Tx/TxHistory/TxEntry.jsx]: 1. Since you're just using the `fromWei` util function, I'd prefer we don't import a `web3` instance. Would recommend using our util function, `weiToEther`, when this PR is merged in: https://github.com/ethereum/ethereum-react-components/pull/76/files#diff-3f81a5326768d781840bf95be050c2c1R19

2. Similary, import and use BN functions from utils, so we can manage BN instances only in that one place.

File: src/components/Tx/TxHistory/TxEntry.jsx [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+    const linkedTxHash = TxEntry.linkedTxHash(tx)
+    const etherAmount = web3.utils.fromWei(tx.value, 'ether')
+    let etherAmountUSD
+
+    if (tx.networkId === 1 && etherPriceUSD) {
+      etherAmountUSD = (etherAmount * etherPriceUSD).toFixed(2)
+    }
+
+    const gasPriceEther = web3.utils.fromWei(tx.gasPrice, 'ether')
+    const gasPriceGwei = web3.utils.fromWei(tx.gasPrice, 'gwei')
```
Comment (marcgarreau) [inline src/components/Tx/TxHistory/TxEntry.jsx]: Ah, we don't have a `weiToGwei` util function yet. Consider adding one for use here, otherwise you'll need to use `weiToEth`, then `ethToGwei`.

File: src/components/Tx/TxHistory/TxEntry.jsx (hunk 1 above)
Comment (ryanio) [reply src/components/Tx/TxHistory/TxEntry.jsx]: 1. This was just to ease the transition, since the component was already coded. I think that all new code being written should have child elements styled as separate JS objects. I can convert this component if it's a good use of time, but thought there might be other priorities.

2. I like the idea of including style configurability, but maybe it can start being added on an "as-needed" basis.

File: src/components/index.js (hunk 4 above)
Comment (ryanio) [reply src/components/index.js]: Ah didn't know, will re-alphabetize

File: src/lib/txHistory.import.less (hunk 5 above)
Comment (ryanio) [reply src/lib/txHistory.import.less]: yes for sure i'm on the same page, forgot to remove the file (used it as a stub to start)

File: src/components/Tx/TxHistory/TxEntry.jsx (hunk 6 above)
Comment (ryanio) [reply src/components/Tx/TxHistory/TxEntry.jsx]: sounds good

File: src/components/Tx/TxHistory/TxEntry.jsx (hunk 7 above)
Comment (ryanio) [reply src/components/Tx/TxHistory/TxEntry.jsx]: sounds good

File: src/components/Tx/TxHistory/TxEntry.jsx (hunk 2 above)
Comment (ryanio) [reply src/components/Tx/TxHistory/TxEntry.jsx]: Did a good amount of reorg and simplification, should be much easier to read the component now. Thanks!

**Comment by ryanio** [comment]
Closes #62
